GoogleForCreators/web-stories-wp
# Fix detecting contenteditable for copy-paste

## Summary

Fixes the behavior where a new text element is created when pasting into an existing text element.



## Relevant Technical Choices



## To-do



## User-facing changes

Pasting into a text element now pastes the text inside it instead of creating a new element


## Testing Instructions
I
1. Add a Text element and go to edit mode.
2. Select part of the text, copy it.
3. Now paste the copied text into the same text element in edit mode.
4. Verify that the text was pasted as expected into the text element.

II
1. Copy some sentence from a text editor (or from anywhere else)
2. Add a text element and go to edit mode.
3. Paste the text into the element.
4. Verify that the text was pasted inside the text element and new element was not created.

III
1. Copy some sentence from a text editor (or from anywhere else)
2. Select the Page
3. Paste
4. Verify that a new text element with the copied text was created.

IV
1. Add a text element
2. Copy the element without going into edit mode
3. Now paste it
4. Verify that a new duplicate element was created.

V
1. Add a text element
2. Copy the element _without_ going into edit mode
3. Add a new text element and go to edit mode
4. Paste
5. Verify that the content of the first text element was pasted into the new element edit mode instead of creating a new element (markup does not have to prevail).




---



Fixes #1953
